src/pages/[platform]/build-a-backend/graphqlapi/connect-from-server-runtime/index.mdx

```diff
@@ -20,18 +20,655 @@ export function getStaticProps(context) {
+
+Choose the correct GraphQL API client based on your Next.js Router (App or Pages) and then their use case:
+
+<BlockSwitcher>
+
+<Block name="App Router">
+
+|Use case|Required GraphQL client|
+|-|-|
+|React Server Component|`generateServerClientUsingCookies()` 🍪|
+|Server Actions|`generateServerClientUsingCookies()` 🍪|
+|Route Handler|`generateServerClientUsingCookies()` 🍪|
+|Middleware|`generateServerClientUsingReqRes()` 🌐|
+
+</Block>
+
+<Block name="Pages Router">
+
+**Pages Router**
+
+|Use case|Required GraphQL client|
+|-|-|
+|Server-side component code|`generateServerClientUsingReqRes()` 🌐|
+|API Route|`generateServerClientUsingReqRes()` 🌐|
+|Middleware|`generateServerClientUsingReqRes()` 🌐|
+
+</Block>
+
+</BlockSwitcher>
+
+### Step 2 - Generate the GraphQL API client for Next.js server runtimes
+
+<BlockSwitcher>
+
+<Block name="generateServerClientUsingCookies() 🍪">
+
+To generate a GraphQL API client for the Next.js server runtime using cookies, you need to provide both your Amplify configuration and the cookies function from Next.js.
+
+```ts
+import amplifyConfig from '@/amplifyconfiguration.json'
+import { cookies } from 'next/headers'
+
+export const cookieBasedClient = generateServerClientUsingCookies({
+  config: amplifyConfig,
+  cookies
+})
+```
+
+<Callout>
+
+We recommend to generate the server API client in a utility file. Then, import the generated client in your Next.js React Server Components, Server Actions, or Route Handlers.
+
+</Callout>
+
+</Block>
+
+<Block name="generateServerClientUsingReqRes() 🌐">
+
+To generate a GraphQL API client for the Next.js server runtime using `NextRequest` and `NextResponse`, you need to only provide your Amplify configuration. When making the individual API requests, you'll need to pass it into an [`runWithAmplifyServerContext`](/[platform]/build-a-backend/server-side-rendering) function to pass in the cookies from request and response variables.
+
+```ts
+import amplifyConfig from '@/amplifyconfiguration.json'
+import { createServerRunner } from '@aws-amplify/adapter-nextjs'
+
+export const { runWithAmplifyServerContext } = createServerRunner({
+  config: amplifyConfig
+})
+
+export const reqResBasedClient = generateServerClientUsingReqRes({
+  config: amplifyConfig,
+})
+```
+
+<Callout>
+
+We recommend to generate the server API client in a utility file. Then, import the generated client in your Next.js Middleware, component's server runtime code, and API Routes.
+
+</Callout>
+
+</Block>
+
+</BlockSwitcher>
+
+### Step 3 - Call GraphQL API using generated server API clients
+
+With the generated server API clients, you can make any GraphQL query or mutation request. Subscriptions are not available within server runtimes.
+
+<BlockSwitcher>
+
+<Block name="generateServerClientUsingCookies() 🍪">
+
+Import the cookie-based GraphQL API's server client in your Next.js React Server Component code and make your GraphQL. 
```

Comment:
@renbran is the apostrophe triggering a code error here?

Reply:
It doesn't and I couldn't understand why it would.